src/main/java/org/reaktivity/nukleus/maven/plugin/internal/generate/StringFlyweightGenerator.java

```diff
@@ -219,8 +219,8 @@ private MethodSpec setStringMethod()
                     .addStatement("byte[] charBytes = value.getBytes(charset)")
                     .addStatement("MutableDirectBuffer buffer = buffer()")
                     .addStatement("int offset = offset()")
-                    .addStatement("buffer.putByte(offset, (byte) charBytes.length)")
-                    .addStatement("buffer.putBytes(offset + 1, charBytes)")
+                    .addStatement("buffer.putShort(offset, (short) charBytes.length)")
```

Comment:
May be you need to pass endianness parameter. putShort(offset, BIG_ENDIAN). See DataFW.

You need to do that for all the usages of putShort() and getShort().